tests/node/index.js

```diff
@@ -0,0 +1,32 @@
+var Fs = require('fs');
+var Path = require('path');
+var Env = require('../..').Env;
+var Fetch = require('../..').fetch;
+
+
+var env = new Env('en-US', Fetch);
+var ctx = env.createContext([
+  Path.join(__dirname + '/fixtures/{locale}/entities.l20n')
+]);
+var langs = [
+  {code: 'es-ES'},
+  {code: 'en-US'},
+  {code: 'fr'}
+];
+
+ctx.fetch(langs).then(init);
+
+function init () {
+  ctx.resolve(langs, 'whatCanIAsk').then(function (value) {
+    console.log('SINGLE VALUE: ', value);
+  });
```

Comment:
Let's move this discussion to another issue, pr or bug.  This pr already has a lot of good things in it and I'd like to land it soon.